lib/order/structure/getMakerTakerOrders.js

```diff
@@ -78,9 +78,16 @@ async function getMakerTakerOrders(api, order) {
    * Remainder of the current order
    * @type {Number}
    */
-  const remainder = orders.reduce((result, o) => result -= fromBaseUnits(o.contract.amount), order.amount);
+  let remainder = orders.reduce((result, o) => result -= fromBaseUnits(o.contract.amount), order.amount);
   const assetPrecision = 1/(10 ** order.asset.decimals);
 
+  if (orders.length > 0 && orders[0].type === 'sell') {
+    if (remainder * order.price < 0.5) {
```

Comment:
Shouldn't this use the amount, not the price? And it should be 500000 micro algos

Reply:
@atrefonas 
If `orders[0].type === sell` that means it is an open Sell Order which means the remainder will be an open buy order. To get the total algo amount for an open buy order we multiply the asa amount by the price of the asa in algo. All values at this step are in normal decimal notation which is why `< 0.5` instead of `<50000`. And we are limiting to `0.5` because we don't allow maker buyorders with a total less than 0.5 algo. All decimal conversion is handled in the compile step after this conditional.

Reply:
What happens if a decimal precision issue in the multiplication causes it to be 0.500001 and then the smart contract fails? Maybe should make it (remainder * order.price < 0.500001) or use big.js or similar

Reply:
That's a good point, I'll make the update.

Reply:
In general be super careful of this when working with JS and TEAL, especially when multiplying and dividing stuff where precision gets lost. This caused a lot of edge cases in the v1 SDK initially.